### DISCUSSION TITLE
Rule to forbid imports?

### DISCUSSION THREAD

**Discussion by lra** [terraform-linters/tflint#1928] [Ideas]
## Introduction

People tent do forget import blocks when they use them, a rule to block imports might be useful.

This would force people to not leave useless imports in the code once they imported their resources.

## Proposal

Add an optional rule to fail when imports exist in the codebase.

## References

- https://developer.hashicorp.com/terraform/language/import

**Comment by wata727** [comment]
We recommend the OPA ruleset for these use cases. Below is an example of a policy that prohibits resource declarations.
https://github.com/terraform-linters/tflint-ruleset-opa/tree/1ad0c612d6b6860afc8e50b58e6ece4f8fac87d6/examples/enforce_modules
